tests/functional/testMultihop.js

```diff
@@ -117,524 +102,501 @@ describe('Server list', function() {
-       await vpn.activate();
-
-       // wait and assert vpn connection
-       await vpn.waitForCondition(async () => {
-         return await vpn.getElementProperty(
-                    generalElements.CONTROLLER_TITLE, 'text') == 'VPN is on';
-       });
-       assert.strictEqual(vpn.lastNotification().title, 'VPN Connected');
-       assert.strictEqual(
-           vpn.lastNotification().message,
-           `Connected to ${currentCountry}, ${currentCity}`);
-
-       // back to main view
-       await vpn.waitForElementAndClick(homeScreen.SERVER_LIST_BUTTON);
-
-
-       // switch from multihop to singlehop
-       await vpn
-           .waitForElementAndClick(
-               homeScreen.selectSingleHopServerView.SINGLEHOP_SELECTOR_TAB)
-               await vpn
-           .waitForElementAndClick(
-               homeScreen.selectSingleHopServerView.BACK_BUTTON)
-
-           // wait and assert vpn connection
-           await vpn.waitForCondition(async () => {
-             return await vpn.getElementProperty(
-                        generalElements.CONTROLLER_TITLE, 'text') == 'VPN is
-     on';
-           });
-       assert.strictEqual(vpn.lastNotification().title, 'VPN Connected');
-       assert.strictEqual(
-           vpn.lastNotification().message,
-           `Connected to ${currentCountry}, ${currentCity}`);
-     });
-  */
-});
+  });
+
+  it('check the countries and cities for multihop exits', async () => {
+    await vpn.waitForQueryAndClick(
+        queries.screenHome.serverListView.MULTIHOP_SELECTOR_TAB.visible());
+    await vpn.waitForQuery(queries.screenHome.STACKVIEW.ready());
+
+    await vpn.wait();
+    await vpn.waitForQueryAndClick(
+        queries.screenHome.serverListView.EXIT_BUTTON.visible());
+    await vpn.waitForQuery(queries.screenHome.STACKVIEW.ready());
+
+    for (let server of servers) {
+      const countryId =
+          queries.screenHome.serverListView.generateCountryId(server.code);
+      await vpn.waitForQuery(countryId.visible());
+
+      await vpn.setQueryProperty(
+          queries.screenHome.serverListView.COUNTRY_VIEW, 'contentY',
+          parseInt(await vpn.getQueryProperty(countryId, 'y')));
+
+      if (currentCountryCode === server.code) {
+        assert(
+            (await vpn.getQueryProperty(countryId, 'cityListVisible')) ===
+            'true');
+      }
+
+      if ((await vpn.getQueryProperty(countryId, 'cityListVisible')) ===
+          'false') {
+        await vpn.clickOnQuery(countryId);
+      }
+
+      for (let city of server.cities) {
+        const cityId = queries.screenHome.serverListView.generateCityId(
+            countryId, city.name);
+        console.log('  Waiting for cityId:', cityId);
+        await vpn.waitForQuery(cityId.visible().prop(
+            'checked',
+            currentCountryCode === server.code &&
+                currentCity === city.localizedName));
+      }
+    }
+  });
+
+  // faulty on wasm -- fast follow to investigate
```

Comment:
but it has been working? (if the tests are currently green then I don't understand why it's faulty on wasm)